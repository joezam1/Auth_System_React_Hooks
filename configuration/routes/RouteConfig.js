const routeConfiguration = Object.freeze({
    homePath:'/',
    authLoginPath:'/auth/login',
    authRegisterPath:'/auth/register',
    authLogoutPath:'/auth/logout',
    privateCustomerDashboardPath:'/private/customer/dashboard',
    privateCustomerOrdersPath:'/private/customer/orders',
    privateCustomerSilverOffersPath:'/private/customer/silverOffers',
    privateCustomerGoldOffersPath:'/private/customer/goldOffers',
    privateCustomerPremiumOffersPath:'/private/customer/premiumOffers',
    privateCustomerPremiumClubMemberPath:'/private/customer/premiumClubMember',
    privateCustomerNewsletterPath:'/private/customer/newsletter',
    privateCustomerManagerOrdersPath:'/private/customer/ManagerOrders',
    privateCustomerContactSuppliersPath:'/private/customer/contact/suppliers',
    storeOrdersPath:'/store/orders',
    storeInvoicesPath:'/store/invoices',
    storeWarehousePath:'/store/warehouse',
    storeAdvertisingPath:'/store/advertising'

});

export default routeConfiguration;