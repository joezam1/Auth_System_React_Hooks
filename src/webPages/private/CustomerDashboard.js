import React from 'react';
import { Link } from 'react-router-dom';
import RouteConfig from '../../../configuration/routes/RouteConfig.js';


export default function CustomerDashboard(){

    return(
        <div className="section-Customer-dashboard">
            Customer Dashboard works ok!
            <br/>
            <Link to={RouteConfig.authLogoutPath}>Logout</Link>
        </div>
    )
}