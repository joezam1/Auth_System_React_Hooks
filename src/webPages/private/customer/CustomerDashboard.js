import React , {useEffect, useState} from 'react';
import { Link } from 'react-router-dom';
import RouteConfig from '../../../../configuration/routes/RouteConfig.js'
import LinkButtonPrivateRedirect from '../../../components/LinkButtonPrivateRedirect.js'

//Test: DONE
export default function CustomerDashboard(){

    useEffect(()=>{

    }, []);


    return(
        <div className="section-Customer-dashboard">
            Customer Dashboard works ok!
            <br/>
            <LinkButtonPrivateRedirect redirectToLocation={RouteConfig.privateCustomerOrdersPath} buttonText=" Go to Orders " />
            <br/>
            <LinkButtonPrivateRedirect redirectToLocation={RouteConfig.privateCustomerSilverOffersPath} buttonText=" Go to Silver  Offers " />

            <br/>
            <Link to={RouteConfig.authLogoutPath}>Logout</Link>
        </div>
    )
}