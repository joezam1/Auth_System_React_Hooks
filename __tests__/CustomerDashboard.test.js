import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import RouteConfig from '../configuration/routes/RouteConfig.js';

import CustomerDashboard from '../src/webPages/private/CustomerDashboard.js';


describe('File: CustomerDashboard.js', function(){
    afterEach(cleanup);
    afterEach(() => {
        jest.clearAllMocks();
    });
    describe('Component CustomerDashboard', function(){
        let container;
        beforeEach(()=>{
            container = render( <MemoryRouter> <CustomerDashboard/>  </MemoryRouter>  );
        });

        test('It should take a snapshot', function(){
            expect(container.asFragment(<CustomerDashboard/>)).toMatchSnapshot();
        });


        test('Logout Link Redirects Correctly', function(){
                const link = container.getByText( 'Logout' );
                const link3 = container.getByRole('link');
                fireEvent.click(link);
                var initialPath = window.location.pathname;
                var pathname = link.pathname;
                expect(pathname).toBe(RouteConfig.authLogoutPath);
                expect(link).toBe(link3);
        });

    });
});