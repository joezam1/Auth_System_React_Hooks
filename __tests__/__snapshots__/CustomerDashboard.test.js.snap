// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File: CustomerDashboard.js Component CustomerDashboard It should take a snapshot 1`] = `
<DocumentFragment>
   
  <div
    class="section-Customer-dashboard"
  >
    Customer Dashboard works ok!
    <br />
    <div
      class="button-private-redirect-section"
    >
      <button
        data-testid="button-private-redirect-id"
        type="button"
      >
          Go to Orders  
      </button>
    </div>
    <div
      class="button-private-redirect-section"
    >
      <button
        data-testid="button-private-redirect-id"
        type="button"
      >
          Go to Silver  Offers  
      </button>
    </div>
    <br />
    <a
      href="/auth/logout"
    >
      Logout
    </a>
  </div>
    
</DocumentFragment>
`;
